pkgs/tools/misc/opentelemetry-collector/releases.nix

```
@@ -0,0 +1,172 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  testers,
+  nixosTests,
+  opentelemetry-collector-builder,
+  pkgs,
+  go,
+  git,
+  cacert,
+  stdenv,
+}:
+let
+  # This is the tool OTEL uses to build their distributions.
+  builder = "${opentelemetry-collector-builder}/bin/ocb";
+
+  # Keep the version in sync with the builder.
+  rev = opentelemetry-collector-builder.src.rev;
+
+  version = lib.removePrefix "cmd/builder/v" rev;
+
+  # This is a weird meta-repo where all the open-telemetry collectors are.
+  src = fetchFromGitHub {
+    owner = "open-telemetry";
+    repo = "opentelemetry-collector-releases";
+    rev = "v${version}";
+    hash = "sha256-wHi3/rby/AA37x3BCbTXlLgRaN4DpATLSXpQb31Jr/o=";
+  };
+
+  # Then from this src, we use the tool to generate some go code, including
+  # the go.mod and go.sum files.
+  #
+  # The output depends on which release.
+  mkDistributionSource =
+    {
+      name,
+      hash,
+    }:
+    stdenv.mkDerivation {
+      inherit name;
+
+      nativeBuildInputs = [
+        cacert
+        git
+        go
+      ];
+
+      inherit src;
+
+      outputHash = hash;
+      outputHashMode = "recursive";
+      outputHashAlgo = if hash == "" then "sha256" else null;
```

```suggestion
      outputHashAlgo = null;
```


we always call this with a hash that is not null.

This conditional logic only made sense in places where we didn't pass an SRI hash and had to explicitly communicate the algo separately.

---

This allows empty hashes to be used to recalculate the FOD. It's less work than replacing the FOD with lib.fakeHash;

---

Ah, ok then. Sorry für the noise.